src/components/gauge/gauge.tsx

```diff
@@ -0,0 +1,373 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Listen,
+  Prop,
+  State,
+  h
+} from "@stencil/core";
+
+import { Component as GxComponent } from "../common/interfaces";
+
+@Component({
+  shadow: false,
+  styleUrl: "gauge.scss",
+  tag: "gx-gauge"
+})
+export class Gauge implements GxComponent {
+  @Element() element: HTMLGxGaugeElement;
+
+  /**
+   * The `gxGaugeDidLoad` event is triggered when component has been rendered completely.
+   */
+  @Event() gxGaugeDidLoad: EventEmitter;
+
+  /**
+   * Property of type Style.
+   * Define if shadow will display or not. Default is disabled.
+   */
+  @Prop() styleShadow = false;
+
+  /**
+   * This property allows selecting the gauge type. The allowed values are `circle` or `line` (defautl).
+   */
+  @Prop() type: "line" | "circle" = "line";
+
+  /**
+   *  Set `true` to display the current value. Default is `false`.
+   *
+   */
+  @Prop() showValue = false;
+
+  /**
+   * The minimum value of the gauge
+   *
+   */
+  @Prop() minValue: number;
+
+  /**
+   * The current value of the gauge
+   *
+   */
+  @Prop() value: number;
+
+  /**
+   * Allows specify the width of the circumference _(When gauge is circle type)_ or the width of the bar _(When gauge is Line type)_ in % relative the component size.
+   *
+   */
+  @Prop() thickness = 10;
+
+  /**
+   * The maximum value of the gauge. _Calculated by Default_
```

Comment:
How is it calculated?

Reply:
I'll add a short explanation about the calculation

Reply:
A more extended explanation was added on commit fc0cab9

Reply:
Much better, thanks!